File: lab-2/src/main/scala/com/hamaddo/labforwardlist/LabApplication.scala
package com.hamaddo.labforwardlist

import scalafx.Includes._
import scalafx.application.JFXApp3
import scalafx.scene.Scene
import scalafxml.core.{FXMLView, NoDependencyResolver}

import java.io.IOException

object LabApplication extends JFXApp3 {
  override def start(): Unit = {

    val resource = getClass.getResource("main-view.fxml")
    if (resource == null) {
      throw new IOException("Cannot load resource: main-view.fxml")
    }

    val root = FXMLView(resource, NoDependencyResolver)

    stage = new JFXApp3.PrimaryStage() {
      title = "Lab Forward List"
      scene = new Scene(root)
      resizable = false
    }
  }
}

File: lab-3/src/main/kotlin/ru/nstu/lab3/LabApplication.kt
package ru.nstu.lab3

import javafx.application.Application
import javafx.fxml.FXMLLoader
import javafx.scene.Scene
import javafx.stage.Stage
import java.io.IOException
import ru.nstu.lab3.ListPerformanceTest

class LabApplication : Application() {
    @Throws(IOException::class)
    override fun start(stage: Stage) {
        val fxmlLoader = FXMLLoader(LabApplication::class.java.getResource("main-view.fxml"))

        val scene = Scene(fxmlLoader.load(), 460.0, 440.0)

        stage.title = "Lab Forward List"
        stage.scene = scene
        stage.isResizable = false
        stage.show()
    }
}

fun main() {
    ListPerformanceTest.test(100000)
    ListPerformanceTest.test(150000)
    ListPerformanceTest.test(200000)
    ListPerformanceTest.test(250000)
    ListPerformanceTest.test(300000)
    ListPerformanceTest.test(350000)
    ListPerformanceTest.test(400000)
    ListPerformanceTest.test(450000)
    ListPerformanceTest.test(500000)

    //Application.launch(LabApplication::class.java)
}